utils/channel_access.py

```diff
@@ -124,6 +124,26 @@ def assert_that_pv_is_an_integer_between(self, pv, min, max, timeout=None):
 
         raise AssertionError(error_message)
 
+    def assert_that_pv_is_close(self, pv, expected_value, delta=1e-6, timeout=None):
+        """
+        Assert that the pv has one of the expected values or that it becomes one of the expected value within the
+        timeout
+
+        :param pv: pv name
+        :param expected_value: value to compare to
+        :param delta: maximum acceptable absolute difference between the two values
+        :param timeout: if it hasn't changed within this time raise assertion error
+        :return:
+        :raises AssertionError: if value does not become within tolerance of the requested value
+        :raises UnableToConnectToPVException: if pv does not exist within timeout
+        """
+        error_message = self._wait_for_pv_lambda(lambda: self._value_is_close(pv, expected_value, delta), timeout)
+
+        if error_message is None:
+            return
+
+        raise AssertionError(error_message)
```

Comment:
If possible, `return` should be the last thing a function does. Sometimes we sneakily do it early but it's not especially good practice. This would be better as:

```
if error_message is not None:
    raise ...
```

You then don't even need the `return` statement as it's implcit